pkg/operator2/secret.go

```diff
@@ -17,14 +17,35 @@ func (c *osinOperator) expectedSessionSecret() *v1.Secret {
 	secret, err := c.secrets.Secrets(targetName).Get(targetName, metav1.GetOptions{})
 	if err != nil || isInvalidSessionSecret(secret) {
 		glog.V(4).Infof("failed to get secret %s: %v", targetName, err)
-		return randomSessionSecret()
+		generatedSessionSecret := randomSessionSecret()
+		if isInvalidSessionSecret(generatedSessionSecret) {
+			panic(err) // should never happen
+		}
+		return generatedSessionSecret
 	}
 	return secret
 }
 
 func isInvalidSessionSecret(secret *v1.Secret) bool {
-	// TODO add validation
-	return false
+	// TODO add more validation?
+	//Authentication and Encryption must be same length
+	var sessionSecretsBytes [][]byte
+	for _, v := range secret.Data {
+		sessionSecretsBytes = append(sessionSecretsBytes, v)
+	}
+	for _, ss := range sessionSecretsBytes {
+		var sessionSecrets *legacyconfigv1.SessionSecrets
+		err := json.Unmarshal(ss, &sessionSecrets)
+		if err != nil {
+			return false
+		}
+		for _, s := range sessionSecrets.Secrets {
+			if len(s.Authentication) != len(s.Encryption) {
+				return false
+			}
+		}
+	}
+	return true
```

Comment:
I'm testing this validation locally, now, hold off on reviewing for now so i don't feel dumb : ) thanks.